pkgs/by-name/ss/sse2neon/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  fetchFromGitHub,
+  pkg-config,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sse2neon";
+  version = "1.7.0";
+
+  src = fetchFromGitHub {
+    owner = "DLTcollab";
+    repo = "sse2neon";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-riFFGIA0H7e5StYSjO0/JDrduzfwS+lOASzk5BRUyo4=";
+  };
+
+  postPatch = ''
+    # remove warning about gcc < 10
+    substituteInPlace sse2neon.h --replace-fail "#warning \"GCC versions" "// "
+  '';
+
+  nativeBuildInputs = [ pkg-config ];
+
+  dontInstall = true;
+  # use postBuild instead of installPhase, because the build
+  # in itself doesn't produce any ($out) output
+  postBuild = ''
+    mkdir -p $out/lib
+    install -m444 sse2neon.h $out/lib/
+  '';
```

Any reasons against a custom `installPhase`, e.g.?
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm444 sse2neon.h $out/lib/
    runHook postInstall
  '';
```

---

yes. `sse2neon` does not produce any output in the `buildPhase` (It just compiles the tests. The library in itself is just the header file). With no `$out` nix will error out. Therefore `postBuild` makes sure, that there is an `$out`, before the install Phase starts.

---

I see, that makes sense; thanks for the clarification.